sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/implementation/ReactorConnectionCache.java

```diff
@@ -0,0 +1,268 @@
+                // or RejectedExecutionException when attempting an operation that is closed or if the IO
+                // signal is accidentally closed, retrying in these cases as well. This is inherited from
+                // v1 AmqpChannelProcessor that v2 ReactorConnectionCache replaces.
+                // https://github.com/Azure/azure-sdk-for-java/pull/34122 addresses one source of ISE.
+                // Continue to log (to detect any other unknown edge-case) and recover on ISE.
+                final boolean shouldRetry = error instanceof TimeoutException
+                    || (error instanceof AmqpException && ((AmqpException) error).isTransient()
+                    || (error instanceof IllegalStateException)
+                    || (error instanceof RejectedExecutionException));
+
+                if (!shouldRetry) {
+                    logger.atWarning()
+                        .addKeyValue(TRY_COUNT_KEY, iteration)
+                        .log("Exception is non-retriable, not retrying for a new connection.", error);
+                    return Mono.error(error);
+                }
+
+                final Throwable errorToUse = error instanceof AmqpException
+                    ? error
+                    : new AmqpException(true, "Non-AmqpException occurred upstream.", error, errorContext);
+                // Using the min of retry attempts and max-retries to compute the 'back-off'.
+                // The min is taken so that it never exhaust the retry attempts for transient errors.
+                // This will ensure a new connection will be created whenever the underlying transient error
+                // is resolved. For e.g. when a network connection is lost for an extended period of time and
+                // when the network is restored later, we should be able to recreate a new connection
+                // as long as there is at least one subscriber.
+                final long attempts = Math.min(iteration, retryPolicy.getMaxRetries());
+                final Duration backoff = retryPolicy.calculateRetryDelay(errorToUse, (int) attempts);
+
+                if (backoff == null) {
+                    logger.atWarning()
+                        .addKeyValue(TRY_COUNT_KEY, iteration)
+                        .log("Retry is disabled, not retrying for a new connection.", error);
+                    return Mono.error(error);
+                }
+
+                if (terminated) {
+                    return Mono.error(TERMINATED_ERROR);
+                }
+
+                logger.atInfo()
+                    .addKeyValue(TRY_COUNT_KEY, iteration)
+                    .addKeyValue(INTERVAL_KEY, backoff.toMillis())
+                    .log("Transient error occurred. Retrying.", error);
+
+                return Mono.delay(backoff);
+            }));
+    }
+
+    private static AmqpShutdownSignal closeSignal(String message) {
```

Comment:
Adding a verb to this helps distinguish the purpose.. like getShutdownSignal or createShutdownSignal

Reply:
Agree, renamed to `createShutdownSignal()`

**[DONE]**